File: Array-Vector/Array-Vector_cpp/Sorting/insertion_sort.cpp
#include <bits/stdc++.h>
using namespace std;

void insertionSort(int arr[], int size)
{
    for (int i = 1; i < size; i++)
    {
        int temp = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > temp)
        {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }

    // print array
    for (int i = 0; i < size; i++)
    {
        cout << arr[i] << " ";
    }
}

int main()
{
    int arr[10] = {11, 4, 2, 55, 3, 7, 4, 6, 8, 9};
    int n = sizeof(arr) / sizeof(arr[0]);

    insertionSort(arr, n);
    return 0;
}

File: Array-Vector/Array-Vector_java/Sorting/insertion_sort.java

public class insertion_sort {

    private static void insertionSort(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            int temp = arr[i];
            int j = i - 1;
            while (j >= 0 && temp < arr[j]) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = temp;
        }
    }

    public static void main(String[] args) {
        int[] arr = { 5, 2, 8, 1, 9 };
        insertionSort(arr);

        // print array
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }

}